hs-bindgen/src-internal/HsBindgen/Frontend/Pass/ResolveBindingSpec.hs

```diff
@@ -233,6 +234,36 @@ instance Resolve C.DeclKind where
       C.DeclGlobal ty       -> fmap C.DeclGlobal   <$> resolve ty
       C.DeclConst ty        -> fmap C.DeclConst    <$> resolve ty
 
+instance Resolve C.Reference where
+  resolve (C.ById i)  = pure $ (Set.empty, C.ById i)
+
+instance Resolve C.CommentReference where
+  resolve (C.CommentReference C.Comment{..}) = do
+    x <- C.CommentReference
+       . C.Comment commentCName
+       <$> traverse resolveCommentBlockContent commentChildren
+    pure (Set.empty, x)
+    where
+      resolveCommentInlineContent = \case
+        C.TextContent{..}        -> pure $ C.TextContent {..}
+        C.InlineCommand n r args -> do
+          args' <- traverse ( (\case
+                                Left t  -> pure (Left t)
+                                Right t -> fmap (Right . snd) t
+                              )
+                            . fmap resolve) args
+          pure (C.InlineCommand n r args')
+        C.HtmlStartTag{..}       -> pure $ C.HtmlStartTag{..}
+        C.HtmlEndTag{..}         -> pure $ C.HtmlEndTag{..}
+
+      resolveCommentBlockContent = \case
+        C.Paragraph p              -> C.Paragraph <$> traverse resolveCommentInlineContent p
+        C.BlockCommand n args p    -> C.BlockCommand n args <$> (traverse resolveCommentInlineContent p)
+        C.ParamCommand n i d e p   -> C.ParamCommand n i d e <$> (traverse resolveCommentBlockContent p)
+        C.TParamCommand n p c      -> C.TParamCommand n p <$> (traverse resolveCommentBlockContent c)
+        C.VerbatimBlockCommand{..} -> pure C.VerbatimBlockCommand{..}
+        C.VerbatimLine{..}         -> pure C.VerbatimLine{..}
```

Comment:
Here we probably are better off with a function instead since we always know that the `Set C.NsPrelimDeclId` is always empty for comments.